pandas/core/strings.py

```diff
@@ -742,10 +735,7 @@ def str_split(arr, pat=None, n=None, return_type='series'):
                 n = 0
             regex = re.compile(pat)
             f = lambda x: regex.split(x, maxsplit=n)
-    if return_type == 'frame':
-        res = DataFrame((Series(x) for x in _na_map(f, arr)), index=arr.index)
```

Comment:
isn't this what ``expand=True`` does (and btw, can easily fix #10081) at the same time